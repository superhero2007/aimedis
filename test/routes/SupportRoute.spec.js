import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import SupportRoute from 'routes/SupportRoute';

describe('routes/SupportRoute', () => {
  it('should render the Support Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/support']}>
        <SupportRoute
          exact
          path="/support"
          component={() => (<div>Support</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Support Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/support']}>
        <SupportRoute
          exact
          path="/support"
          component={() => (<div>Support</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
