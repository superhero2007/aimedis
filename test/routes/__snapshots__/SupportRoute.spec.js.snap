// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/SupportRoute should render the Support Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">Support</div>"`;

exports[`routes/SupportRoute should render the Support Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">Support</div>"`;
